### ISSUE TITLE
Memory clean up and allocation in phangorn::coph is not optimal for large simulations

### ISSUE THREAD

**Issue by melmasri** [KlausVigo/phangorn#55]
Hello, 
I have switched from `geiger::cophenetic` to `phangorn::coph`, for the increased speed, specially when used in MCMC. However, I am having a memory allocation problem with phangorn that doesn't exist with geiger code. 

The trash cleaning is not optimal, see the following example. 
```r
rm(list=ls())
require(phangorn)
require(ape)

foo<-function(tree)   b = phangorn:::coph(tree)

tree <- rtree(n =1000)
for(i in 1:30000)  foo(tree)
```
Before the loop R takes about 2.7% of my 4GB memory, and after the loop it is up to 16%. This keeps going as the loop iterations increase. Not convenient in MCMC code with large iterations. See attached images.
![Before](https://user-images.githubusercontent.com/1774554/42582423-17fe03f2-84fd-11e8-8c9a-541c223571cf.png)
![After](https://user-images.githubusercontent.com/1774554/42582430-1ccfb434-84fd-11e8-87ec-dbd8bfb06af1.png)

Partial session info: 

```r
> print(sessionInfo())
R version 3.3.3 (2017-03-06)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 8 (jessie)

other attached packages:
[1] phangorn_2.4.0 ape_5.0       
```

**Comment by KlausVigo** [comment]
Hello @melmasri , 
I added a fix commit 19ed69fc5cdb5c239250ce50f4f0d8b0b1fd32fa , which may has caused a memory leak. 
Your approach of finding memory leaks is not really sophisticated and may be misleading.
You should in this case frequently call the garbage collection `gc()`, possibly in the loop. 
Regards, 
Klaus

**Comment by melmasri** [comment]
Yes you are right. The approach is misleading, I noticed it while running MCMC's, and I wanted a quick replication. I'll use `gc()` for now till you push this to CRAN.  Thanks.
